metadata/metadata.go

```diff
@@ -1848,78 +1977,88 @@ func isResourceReady(res Resource) bool {
 }
 
 type nameStream interface {
-	Recv() (*pb.Name, error)
+	Recv() (*pb.NameRequest, error)
 }
 
 type variantStream interface {
-	Recv() (*pb.NameVariant, error)
+	Recv() (*pb.NameVariantRequest, error)
 }
 
 type sendFn func(proto.Message) error
 
 type initParentFn func(name, variant string) Resource
 
 func (serv *MetadataServer) genericCreate(ctx context.Context, res Resource, init initParentFn) (*pb.Empty, error) {
-	serv.Logger.Info("Creating Generic Resource: ", res.ID().Name, res.ID().Variant)
+	logger := logging.GetLoggerFromContext(ctx).WithValues(map[string]interface{}{"resource-id": res.ID()})
```

Comment:
Shouldn't this use WithResource? or isn't that already added? Confused about this WithValues